cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -24,10 +28,28 @@ EXAMPLES:
 EOUSAGE
 }
 
+# Try to be resilient to SIGPIPE, incase anyone restarts journald while we are running.
+function noncritical {
+  set +e; ${*}; set -e
+}
+
+function checked_mount() {
+  local dev="$1"
+  local location="$2"
+  noncritical echo -n "Mounting: $dev to $location"
+  until grep "^$dev" /etc/mtab > /dev/null; do
+    noncritical sleep 1
+    noncritical echo -n .
+    # mount might think the device is already mounted, so accept nonzero exit status
+    mount "$location" || :
+  done
+  noncritical echo
```

Comment:
This echo is here to insert a newline after the previous `echo -n`. It could fail like any other command in this script that emits to stdout after journald is killed/restarted.

I'll just point out that I think all of the changes in this script that involve `noncritical` are only going to matter if someone recreates the scenario where journald is restarted while dcos_vol_setup is running. The critical changes in this PR avoid that scenario.